tests/functional/spec/routing_and_skipping_section_dependencies.spec.js

```diff
@@ -0,0 +1,236 @@
+import AgePage from "../generated_pages/new_routing_and_skipping_section_dependencies/age.page";
+import HouseHoldPersonalDetailsSectionSummaryPage from "../generated_pages/new_routing_and_skipping_section_dependencies/household-personal-details-section-summary.page";
+import HouseholdSectionSummaryPage from "../generated_pages/new_routing_and_skipping_section_dependencies/household-section-summary.page";
+import ListCollectorAddPage from "../generated_pages/new_routing_and_skipping_section_dependencies/list-collector-add.page";
+import ListCollectorPage from "../generated_pages/new_routing_and_skipping_section_dependencies/list-collector.page";
+import NamePage from "../generated_pages/new_routing_and_skipping_section_dependencies/name-block.page";
+import PrimaryPersonSummaryPage from "../generated_pages/new_routing_and_skipping_section_dependencies/primary-person-summary.page";
+import ReasonNoConfirmationPage from "../generated_pages/new_routing_and_skipping_section_dependencies/reason-no-confirmation.page";
+import RepeatingAgePage from "../generated_pages/new_routing_and_skipping_section_dependencies/repeating-age.page";
+import RepeatingSexPage from "../generated_pages/new_routing_and_skipping_section_dependencies/repeating-sex.page";
+import SecurityPage from "../generated_pages/new_routing_and_skipping_section_dependencies/security.page";
+import SkipAgePage from "../generated_pages/new_routing_and_skipping_section_dependencies/skip-age.page";
+import SkipConfirmationPage from "../generated_pages/new_routing_and_skipping_section_dependencies/skip-confirmation.page";
+import SkipConfirmationSectionSummaryPage from "../generated_pages/new_routing_and_skipping_section_dependencies/skip-confirmation-section-summary.page";
+import SkipSectionSummaryPage from "../generated_pages/new_routing_and_skipping_section_dependencies/skip-section-summary.page";
+
+import HubPage from "../base_pages/hub.page";
+
+describe("Given the routing and skipping section dependencies questionnaire", () => {
+  beforeEach("Load the survey", () => {
+    browser.openQuestionnaire("test_new_routing_and_skipping_section_dependencies.json");
+  });
+
+  it("When I answer 'No' to skipping the age question, Then in the Primary Person section I am asked my name, age and why I didn't confirm skipping", () => {
+    answerNoToSkipAgeQuestion();
+
+    selectPrimaryPerson();
+    answerAndSubmitNameQuestion();
+    answerAndSubmitAgeQuestion();
+    answerAndSubmitReasonForNoConfirmationQuestion();
+
+    expectPersonalDetailsName();
+    expectPersonalDetailsAge();
+    expectReasonNoConfirmationAnswer();
+  });
+
+  it("When I answer 'Yes' to skipping the age question, Then in the Primary Person section I am asked my name and why I didn't confirm skipping", () => {
```

Comment:
The `Then` should highlight the important parts which to me here is that you are not asked the age question. Even just adding `only`/`just` would help, i.e `Then in the Primary Person section I am only asked my name and why I didn't confirm skipping`. You use `just` in the next test case.